x-pack/plugins/endpoint/public/applications/endpoint/store/managing/middleware.ts

```diff
@@ -5,19 +5,28 @@
  */
 
 import { MiddlewareFactory } from '../../types';
-import { pageIndex, pageSize } from './selectors';
+import {
+  pageIndex,
+  pageSize,
+  isOnManagementPage,
+  hasSelectedHost,
+  uiQueryParams,
+} from './selectors';
 import { ManagementListState } from '../../types';
 import { AppAction } from '../action';
 
 export const managementMiddlewareFactory: MiddlewareFactory<ManagementListState> = coreStart => {
   return ({ getState, dispatch }) => next => async (action: AppAction) => {
     next(action);
+    const state = getState();
     if (
-      (action.type === 'userNavigatedToPage' && action.payload === 'managementPage') ||
+      (action.type === 'userChangedUrl' &&
+        isOnManagementPage(state) &&
+        hasSelectedHost(state) !== undefined) ||
```

Comment:
I think `hasSelectedHost()` returns a `boolean`. Why is the conditional checking for `undefined`?